Cognite.Simulator.Utils/LicenseController.cs

```diff
@@ -0,0 +1,226 @@
+            }
+        }
+        
+        /// <summary>
+        /// Acquires a license by invoking the license acquisition function and sets the license status to held.
+        /// Also updates the release timer to ensure the license is released after a certain period.
+        /// </summary>
+        /// <exception cref="Exception">Thrown when an error occurs while acquiring the license.</exception>
+        public void AcquireLicense() {
+            lock (_lock)
+            {
+                try
+                {
+                    if (_licenseHeld)
+                    {
+                        _logger.LogInformation("License already held, skipping acquisition");
+                        return;
+                    }
+                    _logger.LogInformation("Attempting to acquire license");
+                    _acquireLicenseFunc();
+                    _licenseHeld = true;
+                    UpdateReleaseTimer();
+                    _logger.LogInformation("License acquired successfully");
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error acquiring license");
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Releases the license after the timeout has elapsed. This is called by the timer.
+        /// </summary>
+        private void ReleaseLicenseAfterTimeout(object state)
+        {
+            lock (_lock)
+            {
+                _logger.LogInformation("Attempting to release license at scheduled time {ScheduledTime}", _scheduledReleaseTime.ToString("yyyy-MM-dd HH:mm:ss"));
+                if (DateTime.Now >= _scheduledReleaseTime)
+                {
+                    _releaseLicenseFunc();
+                    _licenseHeld = false;  
+                    _logger.LogInformation("License released successfully");
+                }
+                else
+                {
+                    _logger.LogInformation("License not released - current time {CurrentTime} is before scheduled release time", 
+                        DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Releases the license immediately. This is used externally to reset the internal state when an external process finds out
+        /// that the license is no longer held.
+        /// </summary>
+        public void ReleaseLicenseWithoutCallback() {
+            lock (_lock)
+            {
+                _licenseHeld = false;
+                _lastUsageTime = DateTime.Now;
+                _scheduledReleaseTime = DateTime.Now;
+                _releaseTimer.Change(Timeout.Infinite, Timeout.Infinite);
+                _logger.LogInformation("License released without callback");
+            }
+        }
+
+        private void UpdateReleaseTimer() {
+            _lastUsageTime = DateTime.Now;
```

Comment:
I believe we should update _lastUsageTime only on `EndUsage`. otherwise the logline `License has been held for {Duration}` would be lying